### PR TITLE
[docs] build with Documenter#master (v0.28-DEV)

### PR DESCRIPTION
**Pull request by odow**
Preview link: https://jump.dev/JuMP.jl/previews/PR3094/JuMP.pdf

I've been working on a bunch of changes that will land in Documenter@0.28. They're currently unreleased, so here's a WIP PR to see them in practice. 

- [x] https://github.com/JuliaDocs/Documenter.jl/pull/1932
- [x] https://github.com/JuliaDocs/Documenter.jl/pull/1935
- [x] https://github.com/JuliaDocs/Documenter.jl/pull/1937
- [x] https://github.com/JuliaDocs/Documenter.jl/pull/1938
- [x] https://github.com/JuliaDocs/Documenter.jl/pull/1954
- [x] https://github.com/JuliaDocs/Documenter.jl/pull/1955
- [x] https://github.com/JuliaDocs/Documenter.jl/pull/1956
- [x] https://github.com/JuliaDocs/Documenter.jl/pull/1957

### CODE REVIEW

**Comment by odow** [comment]
Things are a lot better (compare against https://jump.dev/JuMP.jl/dev/JuMP.pdf).

Here are some thoughts of things we could still change:

## The warning color looks gold, rather than yellow

image

is not the same as 

image

## I don't really like the font

Particularly these bullets

image

**Comment by odow** [comment]
Success! We can see the full table. cc @mortenpi 

image

Here's a failure though:

image

`GroupBy` produces text with `{tabular}` inside them.

**Comment by odow** [comment]
@mortenpi muuuch better

image

image

image

**Comment by odow** [comment]
Closing. This can wait for the documenter release.

**Comment by mortenpi** [comment]
I would suggest using `#master` -- I don't have a clear ETA on the release right now.

**Comment by odow** [comment]
> I would suggest using #master

Are more breaking changes expected? I guess the two breaking changes are pretty minor: https://github.com/JuliaDocs/Documenter.jl/blob/master/CHANGELOG.md#version-v0280-unreleased

**Comment by mortenpi** [comment]
I would like to get [this one fixed](https://github.com/JuliaDocs/Documenter.jl/issues/1975), but it _may_ lead to some slightly breaking behavior. Just been a little short on time right now.

**Comment by odow** [comment]
The two main arguments for switching are:

  1) lots of improvements to the PDF
  2) a link to GitHub from the header of each page!

These two changes are quite nice, and I try to stay abreast of the changes in Documenter, so I think I'm willing to live dangerously until the next release.

**Comment by mortenpi** [comment]
I would suggest committing a `docs/Manifest.toml` here though, to fix yourself to a commit. That would take some of the adrenaline out of your life.

File: docs/make.jl [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import Pkg
+Pkg.pkg"add Documenter#master"
```
Comment (odow) [inline docs/make.jl:2]:
```suggestion
Pkg.pkg"add Documenter#0150642"
```

**Comment by odow** [comment]
Perhaps I'll just use a specific commit for now.

**Pull request merged**